### PR TITLE
Update usage of TestSink

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Preparing for https://github.com/aspnet/Logging/pull/804

### CODE REVIEW

File: test/Microsoft.AspNetCore.ResponseCaching.Tests/TestUtils.cs [hunk 1]
```diff
@@ -212,14 +213,17 @@ internal static ResponseCachingContext CreateTestContext(ITestSink testSink)
...
         }
 
-        internal static void AssertLoggedMessages(List<WriteContext> messages, params LoggedMessage[] expectedMessages)
+        internal static void AssertLoggedMessages(IEnumerable<WriteContext> messages, params LoggedMessage[] expectedMessages)
         {
-            Assert.Equal(messages.Count, expectedMessages.Length);
-            for (var i = 0; i < messages.Count; i++)
+            Assert.Equal(messages.Count(), expectedMessages.Length);
+
+            messages.Zip(expectedMessages, (message, expectedMessage) =>
```
Comment (pakrym) [inline test/Microsoft.AspNetCore.ResponseCaching.Tests/TestUtils.cs]: Ew, just ToList it inside this method.

Comment (pakrym) [reply test/Microsoft.AspNetCore.ResponseCaching.Tests/TestUtils.cs]: Or, can we use `Assert.Collection` here?

Comment (pakrym) [reply test/Microsoft.AspNetCore.ResponseCaching.Tests/TestUtils.cs]: Then ToList it, let's not make it too fancy.

Comment (JunTaoLuo) [reply test/Microsoft.AspNetCore.ResponseCaching.Tests/TestUtils.cs]: But LINQ is so awesome.

**Comment by JunTaoLuo** [comment]
:up::date:

**Pull request merged**